### PR TITLE
Switch deployment related alerts to toast notifications

### PR DESCRIPTION
**Pull request by sg00dwin**
Change effected actions:
- deploy
- delete deployment
- edit deployment config
- edit resource limits
- edit health checks
- edit yaml (of replication controller)
- deployment rollback
- add,delete,edit env variables
- cancelled deployment
- delete horizontal pod autoscaler

Related PR https://github.com/openshift/origin-web-console/pull/1567

### CODE REVIEW

File: app/scripts/controllers/edit/deploymentConfig.js [hunk 1]
```diff
@@ -371,23 +372,20 @@ angular.module('openshiftConsole')
 
       DataService.update("deploymentconfigs", $scope.updatedDeploymentConfig.metadata.name, $scope.updatedDeploymentConfig, $scope.context).then(
         function() {
-          AlertMessageService.addAlert({
```
Comment (spadgett) [inline app/scripts/controllers/edit/deploymentConfig.js]: I think you can remove `AlertMessageService` from the dependencies above now.

File: app/scripts/controllers/edit/deploymentConfig.js [hunk 2]
```diff
@@ -371,23 +372,20 @@ angular.module('openshiftConsole')
...
+            type: "success",
+            message: "Deployment config " + $scope.updatedDeploymentConfig.metadata.name + " was successfully updated."
           });
           var returnURL = Navigate.resourceURL($scope.updatedDeploymentConfig);
           $location.url(returnURL);
         },
         function(result) {
           $scope.disableInputs = false;
-          $scope.alerts["save"] = {
+          NotificationsService.addNotification({
```
Comment (spadgett) [inline app/scripts/controllers/edit/deploymentConfig.js]: Need an ID on this that is cleared if the edit succeeds or the user cancels the edit.

File: app/scripts/controllers/edit/healthChecks.js [hunk 3]
```diff
@@ -112,12 +113,9 @@ angular.module('openshiftConsole')
                                  object,
                                  context).then(
                 function() {
-                  AlertMessageService.addAlert({
```
Comment (spadgett) [inline app/scripts/controllers/edit/healthChecks.js]: Remove `AlertMessageService` from the dependencies above.

File: app/scripts/controllers/edit/healthChecks.js [hunk 4]
```diff
@@ -112,12 +113,9 @@ angular.module('openshiftConsole')
...
-                    name: $scope.name,
-                    data: {
+                  NotificationsService.addNotification({
                       type: "success",
                       message: displayName + " was updated."
-                    }
                   });
                   $location.url($scope.resourceURL);
                 },
```
Comment (spadgett) [inline app/scripts/controllers/edit/healthChecks.js]: Should the failure below be a toast as well?

File: app/scripts/controllers/edit/yaml.js [hunk 5]
```diff
@@ -159,13 +160,9 @@ angular.module('openshiftConsole')
                     $scope.updatingNow = false;
                     return;
                   }
-
-                  AlertMessageService.addAlert({
```
Comment (spadgett) [inline app/scripts/controllers/edit/yaml.js]: Remove AlertMessageService from the dependencies above.

File: app/scripts/controllers/setLimits.js [hunk 6]
```diff
@@ -102,12 +103,9 @@ angular.module('openshiftConsole')
               $scope.disableInputs = true;
               DataService.update(resourceGroupVersion, $scope.name, object, context).then(
                 function() {
-                  AlertMessageService.addAlert({
```
Comment (spadgett) [inline app/scripts/controllers/setLimits.js]: Remove AlertMessageService from the dependencies above.

File: app/scripts/controllers/setLimits.js [hunk 7]
```diff
@@ -102,12 +103,9 @@ angular.module('openshiftConsole')
...
-                    name: $scope.name,
-                    data: {
+                  NotificationsService.addNotification({
                       type: "success",
                       message: displayName + " was updated."
-                    }
                   });
                   $location.url($scope.resourceURL);
```
Comment (spadgett) [inline app/scripts/controllers/setLimits.js]: The failure just below should be a toast.

File: app/scripts/services/deployments.js [hunk 8]
```diff
@@ -1,11 +1,11 @@
 'use strict';
 
 angular.module("openshiftConsole")
-  .factory("DeploymentsService", function(APIService, DataService, $filter, $q, LabelFilter){
+  .factory("DeploymentsService", function(APIService, NotificationsService, DataService, $filter, $q, LabelFilter){
     function DeploymentsService() {}
     var annotation = $filter('annotation');
 
-    DeploymentsService.prototype.startLatestDeployment = function(deploymentConfig, context, $scope) {
```
Comment (spadgett) [inline app/scripts/services/deployments.js]: Nice to see `$scope` removed from these functions.

**Comment by jwforres** [comment]
closing since this is replaced by #1646

**Pull request closed without merging**